docs/how-to-guides.md

```diff
@@ -0,0 +1,383 @@
+
+If your library was using [Reason syntax](https://reasonml.github.io/en/) (`re`
+files), you will need to add `"reason"` to the list of dependencies. If the
+library was using ReScript syntax (`res` files), you will need to add
+`rescript-syntax` to the list of dependencies. You can read more about how to
+migrate from ReScript syntax in the section below.
+
+At this point, we can create a [local opam
+switch](https://opam.ocaml.org/blog/opam-local-switches/) to start working on
+our library:
+
+```bash
+opam switch create . 5.1.0~rc2 -y --deps-only
+```
+
+Once this step is done, we can call `dune` from the library folder, but first we
+need some configuration files.
+
+### Add a `dune-project` file
+
+Create a file named `dune-project` in the library root folder. This file will
+tell Dune a few things about our project configuration:
+
+```text
+(lang dune 3.8)
+
+(using melange 0.1)
+```
+
+### Replace the `bsconfig.json` file with one or multiple `dune` files
+
+Now, we need to add a `dune` file where we will tell Dune about our library. You
+can put this new file next to the library sources, it will look something like
+this:
+
+```text
+(library
+ (name lib)
+ (modes melange))
+```
+
+Let's see how the most common configurations in `bsconfig.json` (or
+`rescript.json`) map to `dune` files. You can find more information about these
+configurations in the [Rescript
+docs](https://rescript-lang.org/docs/manual/latest/build-configuration) and in
+the [Dune docs](https://dune.readthedocs.io/en/stable/dune-files.html#library).
+
+#### `name`, `namespace`
+
+These two configurations map to Dune `(wrapped <boolean>)` field in the
+`library` stanza. By default, all Dune libraries are wrapped, which means that a
+single module with the name of the library is exposed at the top level. So e.g.
+of your `bsconfig.json` had `"namespace": false`, you can add `(wrapped false)`
+to your library, although wrapped libraries are heavily encouraged to avoid
+global namespace pollution.
+
+#### `sources`
+
+Dune works slightly differently than ReScript when it comes down to including
+source folders as part of a library.
+
+By default, when Dune finds a `dune` file with a `library` stanza, it will
+include just the files inside that folder to the library itself (unless the
+`modules` field is used). If you want to create a library with multiple
+subfolders in it, you can use the following combination of stanzas:
+
+- `(include_subdirs unqualified)`
+  ([docs](https://dune.readthedocs.io/en/stable/dune-files.html#include-subdirs)):
+  This stanza tells Dune to look for sources in all the subfolders of the folder
+  where the `dune` file lives.
+- `(dirs foo bar)`
+  ([docs](https://dune.readthedocs.io/en/stable/dune-files.html#dirs)): This
+  stanza tells Dune to only look into `foo` and `bar` subdirectories of the
+  current folder.
+
+So for example, if your library had this configuration in its `bsconfig.json`:
+
+```json
+{
+  "sources": ["src", "lib"]
```

Comment:
pick something other than `lib`, since bucklescript uses that for artifacts

Reply:
https://github.com/melange-re/melange-re.github.io/pull/111/commits/1f40d9cbf2f68fca8bd47770107d74845520041a